### PR TITLE
Adapt DNS library.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: lib/net/modules/dns.toit [hunk 1]
```diff
@@ -640,29 +641,30 @@ class SrvResource extends StringResource:
...
+  parts_ reader packet parts
   return parts.join "."
 
-parts_ packet/ByteArray position/int parts/List [position-block] -> none:
-  while packet[position] != 0:
-    size := packet[position]
+parts_ reader/io.Reader packet/ByteArray parts/List -> none:
+  while true:
+    size := reader.peek-byte
+    if size == 0: break
```
Comment (erikcorry) [inline lib/net/modules/dns.toit:656]: Why change the code for zero length parts? Now we don't add a zero length string.

File: lib/net/modules/dns.toit [hunk 2]
```diff
@@ -674,58 +676,58 @@ create-dns-packet queries/List records/List -> ByteArray
...
 
   status-bits := is-response ? 0x8000 : 0x0000
   if is-authoritative: status-bits |= 0x0400
 
-  result.write-int16-big-endian id
-  result.write-int16-big-endian status-bits
-  result.write-int16-big-endian queries.size
-  result.write-int16-big-endian records.size
-  result.write-int32-big-endian 0  // Don't support the other things.
+  result.big-endian.write-int16 id
```
Comment (erikcorry) [inline lib/net/modules/dns.toit:684]: Is it possible to cache result.big-endian in a variable to avoid it being looked up on every single line?

**Review by erikcorry — APPROVED** [review]

File: lib/net/modules/dns.toit (hunk 1 above)
Comment (floitsch) [reply lib/net/modules/dns.toit:656]: as discussed in person. This should still do the same thing.

File: lib/net/modules/dns.toit (hunk 2 above)
Comment (floitsch) [reply lib/net/modules/dns.toit]: done.

**Pull request merged**